config.yaml

```diff
@@ -21,6 +21,7 @@ default:
    --plugin-option=tmpfs:required_ram_mb=39800"
  commit_updates: True
  push_updates: True
+ build_dependencies: True
```

Comment:
I don't think that this global attribute is needed, I'd prefer to have a clean_build_dependencies_cache boolean option instead. If this option is disabled, the dependencies reuse would be dynamic instead of the static approach you suggest here. We would not assume that all dependencies are already pre-built or not; for each package, the build system auto-detects if the dependencies of a package are already built in the system and reuses them if they are. To achieve this, we just need to choose an adequate primary key for a package build (eg `<package name`>-`<commit ID`>) and somehow keep an updated table of the built packages. One way to achieve this is by saving each build package in a result dir of format <primary_key>, later parsing this directory structure to infer if the dependency package was already build or not and reuse the package if it was already built.